internal/providers/azure/azure.go

```diff
@@ -97,19 +108,17 @@ func FetchFromOvfDevice(f *resource.Fetcher, ovfFsTypes []string) (types.Config,
 		)
 	}()
 
+	logger.Debug("checking for config drive")
+	if _, err := os.Stat(filepath.Join(mnt, "ovf-env.xml")); err != nil {
+		return nil, fmt.Errorf("failed to identify config drive: %v", err)
+	}
+
 	logger.Debug("reading config")
 	rawConfig, err := ioutil.ReadFile(filepath.Join(mnt, configPath))
 	if err != nil && !os.IsNotExist(err) {
-		return types.Config{}, report.Report{}, fmt.Errorf("failed to read config: %v", err)
-	}
-
-	return util.ParseConfig(logger, rawConfig)
-}
-
-func waitForCdrom(logger *log.Logger, devicePath string) {
-	for !isCdromPresent(logger, devicePath) {
-		time.Sleep(time.Second)
+		return nil, fmt.Errorf("failed to read config: %v", err)
 	}
+	return rawConfig, nil
 }
 
 func isCdromPresent(logger *log.Logger, devicePath string) bool {
```

Comment:
The debug prints in this function don't specify the `devicePath`, and it's no longer obvious which device we're talking about, so the path should probably be added to the messages.